src/edges_analysis/analysis/beams.py

```diff
@@ -498,16 +498,16 @@ def sky_convolution_generator(
         el_above_horizon = el[horizon_mask]
 
         # Selecting sky data above the horizon
-        sky_above_horizon = np.zeros_like(sky_map)
-        sky_above_horizon = sky_map[horizon_mask,:]
+        sky_above_horizon = sky_map
+        sky_above_horizon[~horizon_mask, :] = 0
```

Comment:
I think you want this to be NaN, not zero. I get the point of setting to zero -- we see "nothing" below the horizon. But the normalization below (`n_pix_ok`) depends on the stuff below horizon being NaN, I think.